server/protocols/jmap/src/test/java/org/apache/james/jmap/model/JmapFlagTest.java

```diff
@@ -0,0 +1,42 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.model;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import javax.mail.Flags;
+
+import org.junit.Test;
+
+public class JmapFlagTest {
+    @Test
+    public void getImapFlagShouldReturnImapFlagOfJmapFlag() throws Exception {
+        assertThat(JmapFlag.$Draft.getImapFlag()).isEqualTo(Flags.Flag.DRAFT);
+    }
+
+    @Test
+    public void fromFlagShouldCreateJmapFlagFromImapFlag() throws Exception {
+        assertThat(JmapFlag.fromFlag(Flags.Flag.DRAFT)).isEqualTo(JmapFlag.$Draft);
+    }
+
+    @Test
+    public void fromFlagShouldBeNullInCaseOfUnknowJmapFlag() throws Exception {
+        assertThat(JmapFlag.fromFlag(Flags.Flag.DELETED)).isNull();
```

Comment:
Oups!!! Maybe return type should be an optional ;-) Or maybe you should throw ;-)